coniql/caplugin.py

```diff
@@ -21,32 +29,79 @@
 
 class CAChannel(Channel):
     def __init__(
-        self,
-        value: AugmentedValue,
-        config: ChannelConfig,
-        meta_value: AugmentedValue,
-        writeable: bool = True,
-        last_channel: "CAChannel" = None,
+        self, name, config: ChannelConfig, initial_value: AugmentedValue,
     ):
-        self.value = value
-        self.meta_value = meta_value
-        self.writeable = writeable
+        self.name = name
         self.config = config
-        self.last_channel = last_channel
-        self.formatter: Optional[ChannelFormatter] = None
+        self.value = initial_value
+        self.meta_value: Optional[AugmentedValue] = None
+        self.precision = 0
+        self.formatter = ChannelFormatter()
+        self.display = ChannelDisplay(
+            description=self.name,
+            role="RW",
+            widget=self.config.widget or Widget.TEXTINPUT,
+            form=self.config.display_form,
+        )
+        self.writeable = True
```

Comment:
How would we get the cainfo information in here?

Reply:
Ah yes. As far as I know that would have to be another request.

Reply:
I guess we could do a `cainfo` when the first update comes in, before starting the metadata_monitor...